### PR TITLE
Fix velocity scaling factor calculations and support multi-DOF joints in Servo

### PR DESCRIPTION
**Pull request by sea-bass**
This PR fixes a safety bug we found with MoveIt Servo, in which the velocity scaling override could cause the robot commands sent to exceed the robot's joint velocity limits. This was discovered because of bad IK solutions "teleporting" the robot in mock hardware.

This correctly takes the minimum scaling factor of the joint safety limits (which MUST be always met) and the override scaling factor, if one is specified.

I also ended up fixing another issue in which Servo was only working for joints with 1DOF, because we were making assumptions about only extracting the first variable from a potentially multi-DOF joint. It's also why the scaling factor unit test was ignoring the first joint, which is a virtual joint in the robot model and not even part of the `panda_arm` joint model group.

@ibrahiminfinite FYI

### CODE REVIEW

**Review by marioprats — APPROVED** [review]

**Comment by sea-bass** [comment]
@marioprats I found another issue with multi-DOF joints so the PR grew in scope. Mind re-reviewing? Thanks!

**Review by eholum — APPROVED** [review]
This makes sense to me - and confirmed it addresses joint jumps issues with mock hw interfaces.

File: moveit_ros/moveit_servo/src/servo_node.cpp [hunk 1]
```diff
@@ -163,7 +163,8 @@ void ServoNode::pauseServo(const std::shared_ptr<std_srvs::srv::SetBool::Request
   else
   {
     // Reset the smoothing plugin with the robot's current state in case the robot moved between pausing and unpausing.
-    servo_->resetSmoothing(servo_->getCurrentRobotState());
+    last_commanded_state_ = servo_->getCurrentRobotState();
```
[the remark this answers is not in the document]
Comment (sea-bass) [reply moveit_ros/moveit_servo/src/servo_node.cpp:166]: yep, I think it was that. When the robot returned to unpaused, it seems there were some states in which not sending a command caused it to go into `smoothHalt` and leap over to a stale `last_commanded_state_`.

**Review by marioprats — COMMENTED** [review]
LGTM, thanks!

**Pull request merged**